### PR TITLE
Add TSC Task Provider

### LINKED ISSUE #26079: Contribute tasks for building TypeScript

### PR DESCRIPTION
**Pull request by mjbvz**
Fixes #26079

Adds a task provider for building typescript projects.

### CODE REVIEW

File: extensions/typescript/package.json [hunk 1]
```diff
@@ -7,6 +7,7 @@
   "license": "MIT",
   "publisher": "vscode",
   "aiKey": "AIF-d9b70cd4-b9f9-4d70-929b-a071c400b217",
+  "enableProposedApi": true,
```
Comment (dbaeumer) [inline extensions/typescript/package.json]: No need to use proposed API here anymore. I moved the API to vscode.d.ts

File: extensions/typescript/src/features/taskProvider.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
+/*---------------------------------------------------------------------------------------------
```
Comment (dbaeumer) [inline extensions/typescript/src/features/taskProvider.ts]: I like the idea of inferring the tsconfig.json to use from the open active TS file. However if no file is open I think we should still provide a reasonable command and not just look into the workspace root. We should search for all tsconfig.json and return all of them. If a build is trigger the task runner will run all of them

Regarding activation: I have a similar problem in ESLint where I need to active the plugin to simply being able to listen to config changes. This is all I do in activate. In addition I listen to open event and if a JS files opens I start the ESLint server by hand. So basically I have an activate and a realActivate. We could do the same for TS and start the tsserver when we get a open event for a TS file in the listener registered in activate.

File: extensions/typescript/package.json (hunk 1 above)
Comment (mjbvz) [inline extensions/typescript/package.json]: I'm still seeing: 

screen shot 2017-05-23 at 6 51 21 pm

Was this a recent change?

Comment (dbaeumer) [inline extensions/typescript/package.json]: My fault. I moved the API but I need to remove a wrapper function. Will do an push.

File: extensions/typescript/src/features/taskProvider.ts (hunk 2 above)
Comment (mjbvz) [inline extensions/typescript/src/features/taskProvider.ts]: Thanks. I've updated the activation as you described and it seems to work well. Will take a look at scanning for `tsconfig.json` outside of the root in a follow up PR

**Pull request merged**